src/draw.c

```diff
@@ -725,9 +727,11 @@ static void render_content(cairo_t *c, struct colored_layout *cl, int width, int
                 cairo_fill(c);
                 // border
                 cairo_set_source_rgba(c, cl->frame.r, cl->frame.g, cl->frame.b, cl->frame.a);
-                // TODO draw_rect instead of cairo_rectangle resulted in blurry lines. Why?
-                cairo_rectangle(c, (frame_x + half_frame_width) * scale, (frame_y + half_frame_width) * scale, (progress_width - frame_width) * scale, progress_height * scale);
-                cairo_set_line_width(c, frame_width * scale);
+                draw_rect(c,
```

Comment:
The regression is here. Before it was using floats directly to cairo_rectangle

Reply:
Oh, good catch! Should we convert draw_rect to floats?

Reply:
I have converted it to double, but this is not enough because of rounding.

Unrelated, but I also noticed I have used `float` while most of the existing code and Cairo are using `double`. So, I have converted everything to `double` (and use of `round()` introduced a conversion to `double` as well).

Reply:
> I have converted it to double, but this is not enough because of rounding.

Yeah it's better then to leave the old cairo_rectangle line that was there. You can go ahead and leave the TODO out though.

> Unrelated, but I also noticed I have used `float` while most of the existing code and Cairo are using `double`. So, I have converted everything to `double` (and use of `round()` introduced a conversion to `double` as well).

:+1: